api/extension-guides/language-model-tutorial.md

```diff
@@ -9,13 +9,324 @@ MetaDescription: Tutorial that walks you through creating a VS Code extension th
+      "title": "Tutor"
+      }
+  ]
+}
+```
+
+We also want a "Tutor" button to show up at the top of the editor so that the user can easily activate the tutor annotations. To do that, add a "menus" section in the "contributes" section, just under the "commands".
+
+```json
+"contributes": {
+  "commands": [
+    {
+      "command": "code-tutor.annotate",
+      "title": "Tutor"
+    }
+  ],
+  "menus": {
+    "editor/title": [
+      {
+        "when": "editorFocus",
+        "command": "code-tutor.annotate",
+        "group": "navigation"
+      }
+    ]
+  }
+}
+```
+
+- The "editor/title" specifies where we want this menu option - which is in the title bar of whichever tab is currently open in the editor area.
+- The "when" specifies when we want this menu option to be shown. In this case, we want it to be shown whenever the editor has focus.
+- The "command" specifies which command we want to run when the user clicks this menu option. In this case, we want to run the "code-tutor.annotate" command that we defined earlier.
+- The "group" specifies where we want this menu option to be shown in the title bar. The "navigation" group is the section that is on the top right of the editor.
+
+<!-- TODO: Add "Group" screenshot -->
+![]()
+
+At this point, you can run the extension from the "Run and Debug" view in the sidebar. This will open a new VS Code instance with the extension installed. If you open a project that contains code files and then open a code file, you should see the "Tutor" button in the title bar of the editor. You'll also see a "Tutor" option in the Command Palette.
+
+However, if you click on the "Tutor" button or select the option from the Command Palette, you'll see an error message that says "Command 'code-tutor.annotate' resulted in an error (command 'code-tutor.annotate' not found)". That's because we haven't implemented the command yet, so let's do that next.
+
+## Step 3: Implement the "Tutor" command
+
+Commands are defined in the `package.json` file, but they are implemented in the `src/extension.ts` file. Open this file and locate the following line.
+
+```ts
+const disposable = vscode.commands.registerCommand('code-tutor.helloWorld', () => {
+```
+
+Remember that we changed this command in the `package.json` file to be `code-tutor.annotate` instead of `code-tutor.helloWorld`. We also want to register a TextEditor command instead of just a command. This will give us a reference to the current text editor.
+
+Modify the line so that it looks like this:
+
+```ts
+const disposable = vscode.commands.registerTextEditorCommand('code-tutor.annotate', async (textEditor: vscode.TextEditor) => {
+```
+
+You can now run the extension again, and this time when you click on the "Tutor" button or select the option from the Command Palette, you should see a message that says "Hello from Code Tutor!".
+
+We're now ready to start building the functionality to generate AI-powered code annotations.
+
+## Step 4
```

Comment:
This step is quite long. Maybe it should be split into two steps:
1. combine getting the code and line numbers and sending the LM API request
2. Parse and display

Or even split it in the three steps you already have?